docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -199,6 +204,28 @@ And the following defaulting:
     ControlPlaneReady bool `json:"controlPlaneReady,omitempty"`
 ```
 
+- Add `KubeadmControlPlane.Spec.UpgradeStrategy` defined as:
+
+```go
+    // UpgradeStrategy is a field to select an upgrade strategy. If field is not present
+    // KCP uses default scale out strategy. Only supported additional upgrade strategy is ScaleIn.
+    // +optional
+    UpgradeStrategy UpgradeStrategyType `json:"upgradeStrategy,omitempty"`
+```
+
+- Add `KubeadmControlPlane.UpgradeStrategyType` struct defined as:
+
+```go
+    // UpgradeStrategyType defines the type of upgrade strategy. This struct is similar to MachineDeploymentStrategy in
+    // https://github.com/kubernetes-sigs/cluster-api/blob/master/api/v1alpha3/machinedeployment_types.go
+    // Struct allows addition of the new upgrade strategies and extensions to existing ones.  
+    type UpgradeStrategyType struct {
+      // Type of upgrade strategy. Only supported upgrade strategy is ScaleIn.
+      // +optional
+      Type string `json:"type,omitempty"`
+    }
+```
```

Comment:
AFAIK "Modifications required to existing API Types" documents the changes to other APIs when KCP was introduced, instead here we are documenting an evolution of the KCP types.
Accordingly, I think that this should go into "New API Types", probably in a subchapter named "changes for supporting different upgrade strategies"